### PR TITLE
Removes Firefox search-engine link that throws 404

### LINKED ISSUE #306: Firefox Search Engine plugin link returns 404 error
Under the key resources section: 
https://devguide.python.org/#key-resources

Link to `Firefox search engine plug-in` pointing to https://www.python.org/dev/searchplugin/ appears to be broken.

### PR DESCRIPTION
**Pull request by CuriousLearner**
Resolves #306 

cc @brettcannon

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -265,7 +265,7 @@ Key Resources
     * `Meta tracker <http://psf.upfronthosting.co.za/roundup/meta>`_ (issue
       tracker for the issue tracker)
     * :doc:`experts`
-    * `Firefox search engine plug-in`_
+    * Firefox search engine plug-in
```
Comment (ezio-melotti) [inline index.rst]: If the plug-in is gone, then there's no reason to keep this line around.

Comment (CuriousLearner) [reply index.rst]: Oh, @brettcannon suggested on the issue that the reference should be removed. 

So, what should I do?

Comment (ezio-melotti) [reply index.rst]: Both this line and the link should be removed.

Comment (CuriousLearner) [reply index.rst]: @ezio-melotti fixed :)

**Comment by CuriousLearner** [comment]
Thanks @ezio-melotti 👍

**Pull request merged**